ui/js/models/resultsets_store.js

```diff
@@ -307,19 +307,45 @@ treeherder.factory('ThResultSetStore', [
         };
 
         var addRunnableJobs = function(repoName, resultSet) {
-            return ThRunnableJobModel.get_list(repoName).then(function(jobList) {
-                var id = resultSet.id;
-                _.each(jobList, function(job) {
-                    job.result_set_id = id;
-                    job.id = thAggregateIds.escape(job.result_set_id + job.ref_data_name);
-                });
-
-                if (jobList.length === 0) {
-                    resultSet.isRunnableVisible = false;
-                    thNotify.send("No new jobs available");
+            // first check whether the Gecko Decision Task has completed
+            var tcURLPromise;
+            var platform = _.find(resultSet.platforms, {
+                "name": "gecko-decision",
+                "groups": [{"jobs": [{"state": "completed"}]}]});
+            if (platform) {
+                // Gecko Decision Task has been completed.
+                // Let's fetch the URL of full-tasks-graph.json
+                var job_guid = platform.groups[0].jobs[0].job_guid;
+                tcURLPromise = ThJobDetailModel.getJobDetails(job_guid, {timeout: null});
+            }
+            if (!tcURLPromise) {
+                // Here we are passing false to the results instead of skipping the promise
+                tcURLPromise = $q.when(false);
+            }
+            return tcURLPromise.then(function(results) {
+                // Since we want to try this only on try first
+                var decisionTaskID = "";
+                if (results && repoName === "try") {
+                    decisionTaskID = results[0].url.substring(results[0].url.indexOf("#") + 1);
+                    // Removing last two characters /0
+                    decisionTaskID = decisionTaskID.substring(0, decisionTaskID.lastIndexOf('/'));
                 }
+                // Will be used later for the GET request
+                resultSet.geckoDecisionTaskID = decisionTaskID;
+                return ThRunnableJobModel.get_list(repoName, {"decisionTaskID": ""}).then(function(jobList) {
```

Comment:
@camd this is the UI restriction for now. Once action tasks are ready, we can make this `decisionTaskID`